### PR TITLE
fix: resource lifecycle

### LINKED ISSUE #2352: the number of workspace members dont decrease after user has been deleted
we remove the user from page "access/accounts"

### PR DESCRIPTION
**Pull request by wansir**
Signed-off-by: hongming <talonwan@yunify.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Release resources after the user is deleted.

**Which issue(s) this PR fixes**:

Fixes #2352

**Special notes for reviewers**:
```
```

**Additional documentation, usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/controller/user/user_controller.go [hunk 1]
```diff
@@ -483,6 +489,59 @@ func (c *Controller) ldapSync(user *iamv1alpha2.User) error {
 	}
 }
 
+func (c *Controller) deleteRoleBindings(user *iamv1alpha2.User) error {
+	listOptions := metav1.ListOptions{LabelSelector: labels.SelectorFromSet(labels.Set{iamv1alpha2.UserReferenceLabel: user.Name}).String()}
+	if result, err := c.ksClient.IamV1alpha2().GlobalRoleBindings().List(listOptions); err != nil {
```
Comment (zryfish) [inline pkg/controller/user/user_controller.go:494]: No need to check the existence first, just issue the delete request, ignore the NotFoud error.

Comment (wansir) [reply pkg/controller/user/user_controller.go:494]: The logic like `kubectl delete <resources> -l  <key>=<value>`.

File: pkg/controller/user/user_controller.go [hunk 2]
```diff
@@ -483,6 +523,88 @@ func (c *Controller) ldapSync(user *iamv1alpha2.User) error {
 	}
 }
 
+func (c *Controller) deleteRoleBindings(user *iamv1alpha2.User) error {
+
```
Comment (zryfish) [inline pkg/controller/user/user_controller.go:527]: This doesn't improve the situation, please use `DeleteCollection` in clientset, it's more efficient

Comment (wansir) [reply pkg/controller/user/user_controller.go]: fixed

**Comment by zryfish** [comment]
Unit test failed, please have a look

**Pull request merged**